server/src/main/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeAction.java

```diff
@@ -252,63 +257,75 @@ protected void doStart(ClusterState clusterState) {
-                                        );
-                                        retryOnMasterChange(clusterState, cause);
-                                    } else {
-                                        listener.onFailure(exp);
+
+                        if (clusterManagerNode.getVersion().onOrAfter(Version.V_2_11_0) && checkTermVersion()) {
+                            transportService.sendRequest(
+                                clusterManagerNode,
+                                ClusterTermVersionAction.NAME,
+                                new ClusterTermVersionRequest(),
+                                new TransportResponseHandler<ClusterTermVersionResponse>() {
+                                    @Override
+                                    public void handleResponse(ClusterTermVersionResponse response) {
+                                        if (response.getTerm() == clusterState.term() && response.getVersion() == clusterState.version()) {
+                                            // same as local execute
+                                            threadPool.executor(executor)
+                                                .execute(
+                                                    ActionRunnable.wrap(
+                                                        GetDelegateForLocalExecute(clusterState),
+                                                        l -> clusterManagerOperation(task, request, clusterState, l)
+                                                    )
+                                                );
+                                        } else {
+                                            // cluster-manager has updated state
+                                            executeOnClusterManager(nodes.getClusterManagerNode(), clusterState);
+                                        }
+                                    }
+
+                                    @Override
+                                    public void handleException(TransportException exp) {
+                                        handleTransportException(nodes.getClusterManagerNode(), clusterState, exp);
+                                    }
+
+                                    @Override
+                                    public String executor() {
+                                        return ThreadPool.Names.SAME;
+                                    }
+
+                                    @Override
+                                    public ClusterTermVersionResponse read(StreamInput in) throws IOException {
+                                        return new ClusterTermVersionResponse(in);
                                     }
                                 }
-                            }
-                        );
+                            );
+                        } else {
+                            // should be always executed on cluster-manager
```

Comment:
This is a code smell in two parts
1. lets invert the condition to reduce indenting
2. Use a local boolean to name the check something like `shouldExecuteOnClusterManger` then you can delete the comment and be sure its always up to date

Reply:
1. Refactored the code block to a seperate method to reduce indentation.
2. Introduced local variable.